terraform/modules/erouska/efgs.tf

```diff
@@ -0,0 +1,52 @@
+locals {
+  uploadkeystoefgs_invoker_roles = [
+    "roles/cloudfunctions.serviceAgent",
+    "roles/iam.serviceAccountUser"
+  ]
+
+  uploadkeystoefgs_roles = [
+    "roles/cloudfunctions.serviceAgent",
+    "roles/datastore.user"
+  ]
```

Comment:
Not sure about role for reading SQL database, but
1. You also need `"roles/secretmanager.secretAccessor"`.
1. You don't need to access Firestore